be/src/exec/pipeline/sink/sink_io_buffer.h

```diff
@@ -84,10 +84,12 @@ class SinkIOBuffer {
     virtual void cancel_one_sinker() { _is_cancelled = true; }
 
     virtual void close(RuntimeState* state) {
-        _is_finished = true;
         if (_exec_queue_id != nullptr) {
             bthread::execution_queue_stop(*_exec_queue_id);
+            bthread::execution_queue_join(*_exec_queue_id); // make sure the stop task has been executed
```

Comment:
had better to handle the return code of these two function `stop` and `join`

Reply:
added error logging